<?xml version="1.0"?>
<launch>

  <node pkg="nodelet" type="nodelet" name="pcl_manager"  args="manager" output="screen"/>
  
  <!-- Run amplitude filter --> 
  <node pkg="nodelet" type="nodelet" name="amplitude_confidence_filter_nodelet" args="load cob_env_model/AmplitudeConfidenceFilter pcl_manager" output="screen">
  	<remap from="point_cloud2" to="/swissranger/pointcloud2_raw"/>
  	<remap from="point_cloud2_filtered" to="/cob_env_model/point_cloud2_filtered_amp"/>
  	<param name="filter_by_amplitude" type="bool" value="true"/>
  	<param name="filter_by_confidence" type="bool" value="true"/>
	<param name="amplitude_min_threshold" type="double" value="3000"/>
	<param name="amplitude_max_threshold" type="double" value="70000"/>
	<param name="confidence_threshold" type="double" value="62000" />
  </node>  
  
  <!-- Run speckle filter -->
  <node pkg="nodelet" type="nodelet" name="speckle_filter_nodelet" args="load cob_env_model/SpeckleFilter pcl_manager" output="screen">
  	<remap from="point_cloud2" to="/cob_env_model/point_cloud2_filtered_amp"/>
  	<remap from="point_cloud2_filtered" to="/cob_env_model/point_cloud2_filtered_speckle"/>
    <param name="filter_speckle" type="bool" value="true"/> 
  </node>  
  
 <!-- Run tearoff edges filter  -->
  <node pkg="nodelet" type="nodelet" name="tearoff_filter_nodelet" args="load cob_env_model/TearoffFilter pcl_manager" output="screen">
  	<remap from="point_cloud2" to="/cob_env_model/point_cloud2_filtered_speckle"/>
  	<remap from="point_cloud2_filtered" to="/cob_env_model/point_cloud2_filtered_tearoff"/>
	<param name="filter_tearoff" type="bool" value="false"/>
  </node> 
  
  <!-- Run StatisticalOutlierRemoval filter -->
  <node pkg="nodelet" type="nodelet" name="outlier_removal" args="load pcl/StatisticalOutlierRemoval pcl_manager" output="screen">
    <remap from="~input" to="/cob_env_model/point_cloud2_filtered_tearoff"/>
    <rosparam>
      mean_k: 30
      stddev: 0.5
      negative: False
    </rosparam> 
  </node>

</launch>